sympy/concrete/summations.py

```diff
@@ -171,9 +171,35 @@ def _eval_is_zero(self):
         # cancel out. This only answers whether the summand is zero; if
         # not then None is returned since we don't analyze whether all
         # terms cancel out.
-        if self.function.is_zero:
+        if self.function.is_zero or self.has_empty_sequence:
             return True
 
+    def _eval_is_extended_real(self):
+        if self.has_empty_sequence:
+            return True
+        return self.function.is_extended_real
+
+    def _eval_is_positive(self):
+        if self.has_finite_limits and self.has_reversed_limits is False:
+            return self.function.is_positive
+
+    def _eval_is_nonnegative(self):
+        if self.is_zero:
+            return True
```

Comment:
In some situations checking one assumption from another like this can lead to strange bugs to do with assumptions caching. What can happen is that the user asks `if e.is_zero` and then `_ask` ends up here because nonnegative=False implies zero=False. This asks for `is_zero` again but receives the `None` that was temporarily stored there and then you get different results from `_eval_is_nonnegative` than you would have had if the user had asked directly for `e.is_nonnegative`.

Because of the above I would say that a good rule of thumb is that `x._eval_is_Y` should avoid checking `x.is_Z`. Checking properties on `x.args` is fine though. Since the two lines above are not needed (`_ask` will check `_eval_is_zero` later if this returns inconclusive) I would just remove them.

Reply:
I changed it so call _eval_is_zero, to avoid duplicating code. Not sure if it is always the case that all information in there can be derived from the other assumptions.